opta/nice_subprocess.py

```diff
@@ -22,79 +24,133 @@ def nice_run(  # type: ignore # nosec
-            raise ValueError("stdin and input arguments may not both be used.")
-        kwargs["stdin"] = PIPE
-
-    if capture_output:
-        if kwargs.get("stdout") is not None or kwargs.get("stderr") is not None:
-            raise ValueError(
-                "stdout and stderr arguments may not be used " "with capture_output."
-            )
-        kwargs["stdout"] = PIPE
-        kwargs["stderr"] = PIPE
-
-    with Popen(*popenargs, **kwargs) as process:
-        try:
-            stdout, stderr = process.communicate(input, timeout=timeout)
-        except TimeoutExpired as exc:
-            process.send_signal(signal.SIGINT)
-            # Wait again, now that the child has received SIGINT, too.
-            process.wait(timeout=exit_timeout)
-            process.kill()
-            if _mswindows:
-                # Windows accumulates the output in a single blocking
-                # read() call run on child threads, with the timeout
-                # being done in a join() on those threads.  communicate()
-                # _after_ kill() is required to collect that and add it
-                # to the exception.
-                exc.stdout, exc.stderr = process.communicate()
-            else:
-                # POSIX _communicate already populated the output so
-                # far into the TimeoutExpired exception.
-                process.wait()
-            raise
-        except KeyboardInterrupt:
-            print("Received keyboard interrupt")
-            # Wait again, now that the child has received SIGINT, too.
-            process.wait(timeout=exit_timeout)
-            raise
-        except Exception:  # Including KeyboardInterrupt, communicate handled that.
-            process.kill()
-            # We don't call process.wait() as .__exit__ does that for us.
-            raise
-        retcode = process.poll()
-        if check and retcode:
-            raise CalledProcessError(retcode, process.args, output=stdout, stderr=stderr)
-    return CompletedProcess(process.args, retcode or 0, stdout, stderr)
+
+    try:
+        Path("/tmp/optainput.tmp").touch()
+        if input:
+            with open("/tmp/optainput.tmp", "wb") as f:
+                f.write(input)  # type: ignore
+        listargs = list(popenargs)
+        listargs[0].insert(0, "exec")
+        popenargs = tuple(listargs)
+        result = run(
+            *popenargs,
+            input=open("/tmp/optainput.tmp", "rb"),
+            timeout=timeout,
+            check=check,
+            tee=tee,
+            capture_output=capture_output,
+            **kwargs,
+        )
+        os.remove("/tmp/optainput.tmp")
+    except TimeoutError as exc:
+        print("Timeout while running command")
+        signal_all_child_processes()
+        raise exc
+    except KeyboardInterrupt as k:
+        print("Received keyboard interrupt")
+        # Wait again, now that the child has received SIGINT, too.
+        raise k
```

Comment:
```suggestion
        raise
```

Use short form of `raise` if we are re-raising the same exception without modification.